src/tools/perf/lib/ucp_tests.cc

```diff
@@ -590,6 +590,11 @@ class ucp_perf_test_runner {
         rkey        = m_perf.ucp.rkey;
         sn          = 0;
 
+        if (CMD == UCX_PERF_CMD_PUT) {
```

Comment:
pls add a new method "reset_buffers()" to wrap the code in 532-537 (but also reset the receive buffer) and use it here. the code in 532-537 is GPU-safe.